fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -209,6 +209,7 @@ abstract class BaseBrowserFragment :
         get() = _browserToolbarView!!
 
     internal lateinit var browserToolbar: BrowserToolbar
+    private var bottomToolbarContainerView: BottomToolbarContainerView? = null
```

Comment:
For this view, I think you may want to handle this similarly to `BrowserToolbarView` using the getters. I know from https://github.com/mozilla-mobile/fenix/pull/8619/files#diff-c05ae0c657bc8e08d523b61c5a6b7859d170b9fca19785593ea2ebeaefc72b66R102 that this was to prevent memory leaks (this was live-coded in https://www.youtube.com/watch?v=kHHOhPPRytc). 

I suspect `browserToolbar` may also be leaking memory since it's holding the view of `BrowserToolbarView`.

Reply:
```suggestion
    internal var _bottomToolbarContainerView: BottomToolbarContainerView? = null
    
    @VisibleForTesting
    internal val bottomToolbarContainerView: BottomToolbarContainerView
        get() = _bottomToolbarContainerView!!
```
This is the change I am referring to. One thing to note is that we might want to add tests for this, otherwise, we should change `internal` to `private`

Reply:
addressed this, but it was more of a style problem because `bottomToolbarContainerView` reference was being cleared during onDestroy. but after some testing a found two leaks 🤦‍♂️ (`appStore.flowScope` and `browserToolbar`). addressed them

thanks for attaching the video, picked up a few things👍